src/test/java/org/threeten/extra/chrono/TestInternationalFixedChronology.java

```diff
@@ -0,0 +1,1191 @@
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.threeten.extra.chrono;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.time.DateTimeException;
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.Month;
+import java.time.Period;
+
+import java.time.chrono.ChronoPeriod;
+import java.time.chrono.Chronology;
+import java.time.chrono.Era;
+import java.time.chrono.IsoEra;
+
+import java.time.temporal.ChronoField;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalAdjusters;
+import java.time.temporal.TemporalField;
+import java.time.temporal.TemporalUnit;
+import java.time.temporal.UnsupportedTemporalTypeException;
+import java.time.temporal.ValueRange;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+/**
+ * Test.
+ */
+@Test
+@SuppressWarnings({ "static-method", "javadoc" })
+public class TestInternationalFixedChronology {
+
+    //-----------------------------------------------------------------------
+    // Chronology.of(String)
+    // Chronology.getId
+    // Chronology.getCalendarType
+    //-----------------------------------------------------------------------
+    @Test
+    public void test_chronology() {
+        Chronology chronology = Chronology.of("Ifc");
+        assertNotNull(chronology);
+        assertEquals (chronology, InternationalFixedChronology.INSTANCE);
+        assertEquals(chronology.getId(), "Ifc");
+        assertEquals (chronology.getCalendarType(), "ifc");
+    }
+
+    //-----------------------------------------------------------------------
+    // InternationalFixedDate.of
+    //-----------------------------------------------------------------------
+    @DataProvider(name = "samples")
+    Object[][] data_samples() {
+        return new Object[][] {
+                { InternationalFixedDate.of(1, 1, 1), LocalDate.of(1, 1, 1) },
+                { InternationalFixedDate.of(1, 1, 2), LocalDate.of(1, 1, 2) },
+
+                { InternationalFixedDate.of(1, 1, 27), LocalDate.of(1, 1, 27) },
+                { InternationalFixedDate.of(1, 1, 28), LocalDate.of(1, 1, 28) },
+                { InternationalFixedDate.of(1, 2,  1), LocalDate.of(1, 1, 29) },
+                { InternationalFixedDate.of(1, 2,  2), LocalDate.of(1, 1, 30) },
+
+                { InternationalFixedDate.of(1, 13, 28), LocalDate.of(1, 12, 30) },
+                { InternationalFixedDate.of(1, 13, 27), LocalDate.of(1, 12, 29) },
+                { InternationalFixedDate.of(1, false, true), LocalDate.of(1, 12, 31) },
+                { InternationalFixedDate.of(2,  1,  1), LocalDate.of(2,  1,  1) },
+
+                { InternationalFixedDate.of(6, 12, 27), LocalDate.of(6, 12,  1) },
```

Comment:
Year 6 should be switched to 4 (and things adjusted), because the original tests from Pax were about the insertion of the week-long leap-month.